oteljava/context-storage/src/main/scala/org/typelevel/otel4s/oteljava/context/IOLocalContextStorage.scala

```diff
@@ -102,21 +102,21 @@ object IOLocalContextStorage {
                 )
             }
         }
-
-      private def whenPropagationEnabled[A](whenEnabled: => A): F[A] =
-        if (IOLocal.isPropagating) {
-          F.pure(whenEnabled)
-        } else {
-          F.raiseError(
-            new IllegalStateException(
-              "IOLocal propagation must be enabled with: -Dcats.effect.trackFiberContext=true"
-            )
-          )
-        }
     }
 
   /** The method is instrumented by the OTeL Java agent to provide the IOLocal managed by agent.
     */
   private[context] def getAgentLocalContext(): Option[IOLocal[JContext]] = None
 
+  private[oteljava] def whenPropagationEnabled[F[_], A](whenEnabled: => A)(implicit F: MonadCancelThrow[F]): F[A] =
```

Comment:
`ApplicativeThrow` should be enough:
```suggestion
  private[oteljava] def whenPropagationEnabled[F[_], A](whenEnabled: => A)(implicit F: ApplicativeThrow[F]): F[A] =
```